### PR TITLE
Set DNS to rancher for containers in non-rancher network (if special label is set)

### PR DESCRIPTION
**Pull request by alena1108**
https://github.com/rancher/rancher/issues/1669

@ibuildthecloud

### CODE REVIEW

File: events/start_handler.go [hunk 1]
```diff
@@ -22,6 +22,7 @@ const RancherSystemLabelKey = "io.rancher.container.system"
 const RancherIPEnvKey = "RANCHER_IP="
 const RancherNameserver = "169.254.169.250"
 const RancherDomain = "rancher.internal"
+const RancherDns = "io.rancher.dns"
```
Comment (ibuildthecloud) [inline events/start_handler.go]: I'm thinking `io.rancher.network.dns=true`.  @cjellick @vincent99 bikeshedding?

**Comment by ibuildthecloud** [comment]
Besides label name, looks good.

File: events/start_handler.go (hunk 1 above)
Comment (vincent99) [inline events/start_handler.go]: `io.rancher.network.dns` Sounds good to me.... Is there a table of all the special labels and values yet to compare against?

File: events/start_handler.go [hunk 2]
```diff
@@ -92,7 +93,11 @@ func (h *StartHandler) Handle(event *docker.APIEvents) error {
 	if err != nil {
 		return err
 	}
+
 	if rancherIP == "" {
+		if _, ok := c.Config.Labels[RancherDns]; ok {
```
Comment (ibuildthecloud) [inline events/start_handler.go]: Your not checking that it equals "true".  You can just do the following and go will be happy

```go
if c.Config.Labels[RancherDns] == "true" {
...
}
```

**Pull request merged**